File: cookbook/genkit/tinyfn_example.ts
/**
 * TinyFn + Google GenKit: Deterministic math tools for your AI agent.
 *
 * Install: npm install genkit @genkit-ai/mcp @genkit-ai/google-genai
 * Run:     npx tsx tinyfn_example.ts
 */

import { createMcpClient } from "@genkit-ai/mcp";
import { googleAI } from "@genkit-ai/google-genai";
import { genkit } from "genkit";

const TINYFN_API_KEY = process.env.TINYFN_API_KEY!;

const ai = genkit({
  plugins: [googleAI()],
});

const mathClient = createMcpClient({
  name: "tinyfnMath",
  mcpServer: {
    url: "https://api.tinyfn.io/mcp/math/",
    requestInit: {
      headers: {
        "X-API-Key": TINYFN_API_KEY,
      },
    },
  },
});

async function main() {
  await mathClient.ready();

  const tools = await mathClient.getActiveTools(ai);

  const { text } = await ai.generate({
    model: googleAI.model("gemini-2.5-flash"),
    prompt: "What is the factorial of 12?",
    tools,
  });

  console.log(text);
  await mathClient.disable();
}

main();

File: cookbook/openai-agents/tinyfn_example.py
"""TinyFn + OpenAI Agents SDK: Deterministic math tools for your AI agent."""

import asyncio
import os

from agents import Agent, Runner
from agents.mcp import MCPServerStreamableHttp

TINYFN_API_KEY = os.environ["TINYFN_API_KEY"]


async def main():
    async with MCPServerStreamableHttp(
        name="TinyFn Math",
        params={
            "url": "https://api.tinyfn.io/mcp/math/",
            "headers": {"X-API-Key": TINYFN_API_KEY},
        },
        cache_tools_list=True,
    ) as server:
        agent = Agent(
            name="Math Assistant",
            instructions="Use the MCP tools for all calculations. Do not compute anything yourself.",
            mcp_servers=[server],
        )

        result = await Runner.run(agent, "What is the factorial of 12?")
        print(result.final_output)

        result = await Runner.run(agent, "Is 7919 a prime number?")
        print(result.final_output)


if __name__ == "__main__":
    asyncio.run(main())
